### PR TITLE
[RFC] Remove the need for the python to startup Plex

### PR DESCRIPTION
**Pull request by anthonyryan1**
This isn't quite fully baked yet, but it's a step in the correct direction I believe based on the discussions going on at the end of this thread: https://forums.plex.tv/t/when-will-plex-media-server-shift-to-python-3/600689

Bringing it up as a pull request to get the discussion rolling on further changes needed.

Next tasks would include:
* Installing the systemd script
* scanelf: rpath_security_checks() warnings when installing

### CODE REVIEW

**Comment by comio** [comment]
Please can switch to 1.20.2 package?

File: media-tv/plex-media-server/plex-media-server-1.20.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	fowners "${_USERNAME}":"${_USERNAME}" "${default_library_dir}"
+	keepdir "${default_library_dir}"
+
+	# Install the OpenRC init/conf files
+	doinitd "${FILESDIR}/init.d/${PN}"
+	doconfd "${FILESDIR}/conf.d/${PN}"
+
+	# Mask Plex libraries so that revdep-rebuild doesn't try to rebuild them.
+	# Plex has its own precompiled libraries.
+	_mask_plex_libraries_revdep
```
Comment (comio) [inline media-tv/plex-media-server/plex-media-server-1.20.1-r1.ebuild:83]: you need patchelf to fix _codecs_kr.so I think

Comment (comio) [reply media-tv/plex-media-server/plex-media-server-1.20.1-r1.ebuild:83]:
```
+	# Fix RPATH
+	patchelf --force-rpath --set-rpath '$ORIGIN:$ORIGIN/../../../../../../lib' "${ED}"/usr/lib/plexmediaserver/Resources/Python/lib/python2.7/lib-dynload/_codecs_kr.so || die
+

```

File: media-tv/plex-media-server/plex-media-server-1.20.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+	# Install systemd service file
+	systemd_newunit "${FILESDIR}/systemd/${PN}.service" "${PN}.service"
```
Comment (comio) [inline media-tv/plex-media-server/plex-media-server-1.20.1-r1.ebuild:86]:
```
systemd_newunit "${ED}"/usr/lib/plexmediaserver/lib/plexmediaserver.service
```
We already apply a patch to fix PLEX_MEDIA_SERVER_INFO_PLATFORM_VERSION using the Gentoo Profile

**Review by comio — CHANGES_REQUESTED** [review]

**Comment by anthonyryan1** [comment]
I'll get the other changes done in a moment, but the Plex developers said [altering the RPATH isn't correct](https://forums.plex.tv/t/when-will-plex-media-server-shift-to-python-3/600689/24), I've also noticed no issues since we stopped changing the RPATH in my local install.

How did you determine the RPATH modifications need to be re-added?

**Comment by anthonyryan1** [comment]
Closing based on the adapted version merged into the master already.

**Comment by comio** [comment]
> I'll get the other changes done in a moment, but the Plex developers said [altering the RPATH isn't correct](https://forums.plex.tv/t/when-will-plex-media-server-shift-to-python-3/600689/24), I've also noticed no issues since we stopped changing the RPATH in my local install.
> 
> How did you determine the RPATH modifications need to be re-added?

I just copied from other _codecs_*.so:

```
scanelf -r /var/tmp/portage/media-tv/plex-media-server-1.20.2-r1/image/usr/lib/plexmediaserver/Resources/Python/lib/python2.7/lib-dynload/_codecs_jp.so
 TYPE   RPATH FILE 
ET_DYN $ORIGIN:$ORIGIN/../../../../../../lib /var/tmp/portage/media-tv/plex-media-server-1.20.2-r1/image/usr/lib/plexmediaserver/Resources/Python/lib/python2.7/lib-dynload/_codecs_jp.so 
```

**Pull request closed without merging**